File: Heat/Controllers/AddressesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat.Models
Imports Heat.Repositories

Namespace Controllers
    Public Class AddressesController
        Inherits System.Web.Mvc.Controller

        Private db As New HeatDBContext

        ' GET: Addresses
        Function Index() As ActionResult
            Dim address = db.Address.Include(Function(a) a.AddressType)
            Return View(address.ToList())
        End Function

        ' GET: Addresses/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim address As Address = db.Address.Find(id)
            If IsNothing(address) Then
                Return HttpNotFound()
            End If
            Return View(address)
        End Function

        ' GET: Addresses/Create
        Function Create() As ActionResult
            ViewBag.AddressTypeID = New SelectList(db.AddressTypes, "ID", "Description")
            Return View()
        End Function

        <HttpGet> _
        Function Create(customerID As Integer) As ActionResult

        End Function

        ' POST: Addresses/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,AddressTypeID,Street,StreetNumber,City,PostalCode,Province,State,Phone,CellPhone,Fax,Note")> ByVal address As Address) As ActionResult
            If ModelState.IsValid Then
                db.Address.Add(address)
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            ViewBag.AddressTypeID = New SelectList(db.AddressTypes, "ID", "Description", address.AddressTypeID)
            Return View(address)
        End Function

        ' GET: Addresses/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim address As Address = db.Address.Find(id)
            If IsNothing(address) Then
                Return HttpNotFound()
            End If
            ViewBag.AddressTypeID = New SelectList(db.AddressTypes, "ID", "Description", address.AddressTypeID)
            Return View(address)
        End Function

        ' POST: Addresses/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,AddressTypeID,Street,StreetNumber,City,PostalCode,Province,State,Phone,CellPhone,Fax,Note")> ByVal address As Address) As ActionResult
            If ModelState.IsValid Then
                db.Entry(address).State = EntityState.Modified
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            ViewBag.AddressTypeID = New SelectList(db.AddressTypes, "ID", "Description", address.AddressTypeID)
            Return View(address)
        End Function

        ' GET: Addresses/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim address As Address = db.Address.Find(id)
            If IsNothing(address) Then
                Return HttpNotFound()
            End If
            Return View(address)
        End Function

        ' POST: Addresses/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim address As Address = db.Address.Find(id)
            db.Address.Remove(address)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/AddressesController.cs
using System.Linq;
using System.Web.Mvc;
using System.Data.Entity;
using System.Net;
using Heat.Models;
using Heat.Repositories;

namespace Heat.Controllers
{
    public class AddressesController : System.Web.Mvc.Controller
	{
        private IHeatDBContext _db;
        public AddressesController(IHeatDBContext context)
        {
            _db = context;
        }
		
		// GET: Addresses/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return RedirectToAction("badRequest", "error");
			}
			Address address = _db.Addresses.Find(id);
			if ((address == null)) {
				return HttpNotFound();
			}
			return View(address);
		}

		// GET: Addresses/Create
		public ActionResult Create()
		{
			ViewBag.AddressTypeID = new SelectList(_db.AddressTypes, "ID", "Description");
			return View();
		}

		[HttpGet()]
		public ActionResult Create(int customerID)
		{
			return View();
		}

		// POST: Addresses/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,AddressTypeID,Street,StreetNumber,City,PostalCode,Province,State,Phone,CellPhone,Fax,Note")]
Address address)
		{
			if (ModelState.IsValid) {
				_db.Addresses.Add(address);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			ViewBag.AddressTypeID = new SelectList(_db.AddressTypes, "ID", "Description", address.AddressTypeID);
			return View(address);
		}

		// GET: Addresses/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Address address = _db.Addresses.Find(id);
			if ((address == null)) {
				return HttpNotFound();
			}
			ViewBag.AddressTypeID = new SelectList(_db.AddressTypes, "ID", "Description", address.AddressTypeID);
			return View(address);
		}

		// POST: Addresses/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,AddressTypeID,Street,StreetNumber,City,PostalCode,Province,State,Phone,CellPhone,Fax,Note")]
Address address)
		{
			if (ModelState.IsValid) {
				_db.SetModified(address);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			ViewBag.AddressTypeID = new SelectList(_db.AddressTypes, "ID", "Description", address.AddressTypeID);
			return View(address);
		}

		// GET: Addresses/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Address address = _db.Addresses.Find(id);
			if ((address == null)) {
				return HttpNotFound();
			}
			return View(address);
		}

		// POST: Addresses/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			Address address = _db.Addresses.Find(id);
			_db.Addresses.Remove(address);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				_db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
